### PR TITLE
remove hardcoded github mentions

### PR DESCRIPTION
**Pull request by J-Thompson12**
Closes: #666

**What changed?**
Removes remaining hardcoded references to github as the providername.

**Why?**
This makes it easier to add new providers

**How did you test it?**
make sure acceptance tests still work

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: pkg/gitproviders/provider.go [hunk 1]
```diff
@@ -498,6 +500,10 @@ func (p defaultGitProvider) GetCommitsFromOrgRepo(orgRepRef gitprovider.OrgRepos
 	return commits, nil
 }
 
+func (p defaultGitProvider) GetProviderName() GitProviderName {
+	return providerName
```
Comment (jpellizzari) [inline pkg/gitproviders/provider.go:504]: :warning: Global State alarm bells :warning:

What happens when two concurrent HTTP requests try to set this variable at the same time?

**Comment by jpellizzari** [comment]
Does this PR mean that Gitlab will be working, as defined in #666

**Comment by J-Thompson12** [comment]
@jpellizzari gitlab will not work until we have the deploy key stuff in place. Now that we dont look for a token there is no way get it working.

**Comment by jpellizzari** [comment]
@J-Thompson12 The reason I ask is because this PR says it "closes" the issue. Please change it to "Related to"

File: pkg/services/app/add.go [hunk 2]
```diff
@@ -686,7 +685,7 @@ func (a *App) createPullRequestToRepo(info *AppResourceInfo, repo string, appHas
 		return nil
 	}
 
-	userRepoRef := gitproviders.NewUserRepositoryRef(github.DefaultDomain, owner, repoName)
+	userRepoRef := gitproviders.NewUserRepositoryRef(string(a.GitProvider.GetProviderDomain()), owner, repoName)
```
Comment (jpellizzari) [inline pkg/services/app/add.go:688]: Converting to a string here when it looks like this is already a string?

**Review by jpellizzari — APPROVED** [review]

File: pkg/services/app/add.go (hunk 2 above)
Comment (J-Thompson12) [reply pkg/services/app/add.go:688]: yeah that is something left over from the first draft. Sorry I missed that.

**Pull request merged**